### PR TITLE
Save user agent and ip address at order creation time

### PR DESCRIPTION
**Pull request by ds300**
Final part of https://artsyproduct.atlassian.net/browse/PURCHASE-574

Follows on from artsy/metaphysics#1360

I'm assuming we don't want to expose these fields through GraphQL.

Slightly unrelated: it's not obvious to me how certain things are injected into the lexical context like the `context` object itself for graphql resolvers, and `request` in the controllers. Is there somewhere I can read about how this kind of ruby magic works?

### CODE REVIEW

**Comment by ashkan18** [comment]
following up on discussion in https://artsyproduct.atlassian.net/browse/PURCHASE-574, i think for now we can only store it on order creation and remove the update from submission 🙌

**Comment by ashkan18** [comment]
sorry missed that comment, yes context is the object passed to query/mutation resolver mostly used for passing things like authentication tokens and etc. there are some docs [here](http://graphql-ruby.org/authorization/overview.html)

**Comment by ashkan18** [comment]
Nice! thanks!

File: db/migrate/20181030164042_add_order_analytics_fields.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddOrderAnalyticsFields < ActiveRecord::Migration[5.2]
+  def change
+    add_column :orders, :original_user_agent, :string
+    add_column :orders, :original_user_ip, :string
```
Comment (starsirius) [inline db/migrate/20181030164042_add_order_analytics_fields.rb]: Do we need to care about PII?

Comment (alloy) [reply db/migrate/20181030164042_add_order_analytics_fields.rb]: Good point! GDPR does consider it to be personal data https://ec.europa.eu/info/law/law-topic/data-protection/reform/what-personal-data_en

**Pull request merged**